src/plugins/intel_gpu/src/graph/impls/common/condition.cpp

```diff
@@ -36,16 +36,24 @@ struct condition_impl : typed_primitive_impl<condition> {
         auto ev = instance.get_network().get_stream().create_user_event(false);
         set_node_params(instance.get_node());
 
-        bool exec_branch = choose_branch_to_exec(instance);
-        memory::ptr memory_to_copy;
-        if (exec_branch)
-            memory_to_copy = execute_branch(instance.get_net_true(), instance.result_id(), instance.input_memory_ptr());
-        else
-            memory_to_copy = execute_branch(instance.get_net_false(), instance.result_id(), instance.input_memory_ptr());
-        // just copy memory
-        mem_lock<float, mem_lock_type::read> inp_ptr{memory_to_copy, instance.get_network().get_stream()};
-        mem_lock<float, mem_lock_type::write> out_ptr{instance.output_memory_ptr(), instance.get_network().get_stream()};
-        std::copy(inp_ptr.begin(), inp_ptr.end(), out_ptr.begin());
+        auto pred = condition_inst::get_pred_frem_memory(instance.pred_memory_ptr(), instance.get_network().get_stream());
+        network::ptr executed_net = instance.get_inner_networks(pred);
+
+        executed_net->execute({});
+
```

Comment:
We'll need to handle sync for the completion of inner program.
Please refer to the new change in this PR: https://github.com/openvinotoolkit/openvino/pull/17976/files

Reply:
Or using get_output().get_memory() will enable sync.

Reply:
Fixed these codes as I commented offline. please review it again. thanks